src/common/abstractPromote.ts

```diff
@@ -101,6 +102,38 @@ export abstract class PromoteCommand<T extends typeof SfCommand> extends SfComma
     };
   }
 
+  protected async requestPromotionFlow(doceOrg: Org, numRetries: number): Promise<string> {
+    let asyncOperationId: string;
+    try {
+      asyncOperationId = await this.requestPromotion(doceOrg);
+    } catch (error) {
+      const err = error as HttpResponse;
+      // if we get anything besides a 409 we rethrow the error
+      if (err['errorCode'] !== HTTP_CONFLICT_CODE) {
+        throw error;
+      }
+      this.spinner.start('Sync of VCS Events In-Progress');
+      asyncOperationId = await this.retryRequestPromotion(doceOrg, numRetries);
+      this.spinner.stop();
+    }
+    return asyncOperationId;
+  }
+
+  protected async retryRequestPromotion(doceOrg: Org, numRetries: number): Promise<string> {
+    try {
+      const asyncOperationId: string = await this.requestPromotion(doceOrg);
```

Comment:
```suggestion
      return await this.requestPromotion(doceOrg);
```

Minor nit, no sense taking up extra heap space, especially in a recursive method.